synapse/types/state.py

```diff
@@ -584,6 +585,25 @@ def must_await_full_state(self, is_mine_id: Callable[[str], bool]) -> bool:
         # local users only
         return False
 
+    def __contains__(self, key: Any) -> bool:
+        if not isinstance(key, tuple) or len(key) != 2:
+            return False
```

Comment:
```suggestion
            raise TypeError(f"'in StateFilter' requires (str, str) as left operand, not {type(key).__name__}")
```

inspired by `3 in ""`

Reply:
Ah yeah fair. I originally had the key typed with `StateKey`, but then that failed when subclassing from `Mapping`